### PR TITLE
azure-search-documents Swagger Customizations

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR adds Swagger customizations to `azure-search-documents` to perform operations that Swagger transforms can't accomplish.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningDeserializer.java [hunk 1]
```diff
@@ -137,7 +137,9 @@ public Object deserialize(JsonParser jp, DeserializationContext cxt) throws IOEx
 
     @Override
     public void resolve(DeserializationContext cxt) throws JsonMappingException {
-        ((ResolvableDeserializer) this.defaultDeserializer).resolve(cxt);
+        if (this.defaultDeserializer instanceof ResolvableDeserializer) {
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningDeserializer.java:140]: @anuchandy do you know what this method is doing? Wanted to ensure that this change was safe.

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 2]
```diff
@@ -206,18 +206,6 @@
         "new": "method <V> reactor.core.publisher.Flux<V> reactor.core.publisher.Flux<T>::switchOnFirst(java.util.function.BiFunction<reactor.core.publisher.Signal<? extends T>, reactor.core.publisher.Flux<T>, org.reactivestreams.Publisher<? extends V>>, boolean) @ com.azure.core.util.paging.ContinuablePagedFluxCore<C, T, P extends com.azure.core.util.paging.ContinuablePage<C extends java.lang.Object, T extends java.lang.Object>>",
         "justification": "Reactor added new final APIs to Flux."
       },
```
Comment (samvaity) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:208]: Dont need these anymore?

File: sdk/search/azure-search-documents/swagger/pom.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.azure.tools</groupId>
+      <artifactId>azure-autorest-customization</artifactId>
+      <version>1.0.0-beta.3</version>
```
Comment (samvaity) [inline sdk/search/azure-search-documents/swagger/pom.xml:38]: We only have until 1.0.0-beta.2 released?
https://mvnrepository.com/artifact/com.azure.tools/azure-autorest-customization

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json (hunk 2 above)
Comment (alzimmermsft) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json:208]: Yeah, I was thinking of going through the RevApi suppressions in another PR to remove what is no longer relavent.

File: sdk/search/azure-search-documents/swagger/pom.xml (hunk 3 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/swagger/pom.xml:38]: Ah yes, I need to get that released.

cc: @jianghaolu @srnagar

**Pull request merged**